python/ray/tune/tests/test_trial_runner_3.py

```diff
@@ -631,6 +625,25 @@ def testCheckpointAtEndNotBuffered(self):
         os.environ["TUNE_RESULT_BUFFER_LENGTH"] = "7"
         os.environ["TUNE_RESULT_BUFFER_MIN_TIME_S"] = "0.5"
 
+        # Helper class to control runner.step() count.
+        class TrialResultObserver(Callback):
+            def __init__(self):
+                self._counter = 0
+                self._last_counter = 0
+
+            def reset(self):
+                self._last_counter = self._counter
+
+            def just_received_a_result(self):
+                if self._last_counter == self._counter:
+                    return False
+                else:
+                    self._last_counter = self._counter
+                    return True
+
+            def on_trial_result(self, **kwargs):
+                self._counter += 1
+
```

Comment:
Nit: Since this is a duplicate, should we move it to a test utils file or import it from the other test?